### PR TITLE
Introduce openapi spec

### PR DESCRIPTION
**Pull request by dbadura**
Following changes were made:
- add open api spec for internal and external mothership api
- add generating new models using codegen-openapi
- change default output path for charts (optional)
Test infra: https://github.com/kyma-project/test-infra/pull/4145

### CODE REVIEW

File: openapi/api.yaml [hunk 1]
```diff
@@ -0,0 +1,377 @@
+openapi: 3.0.0
+info:
+  title: Reconciler mothership external API
+  version: 1.0.0
+servers:
+  - url: localhost:8080
```
Comment (rJankowski93) [inline openapi/api.yaml:6]:
```suggestion
  - url: http://{host}:{port}/{version}
    variables:
      host:
        default: localhost
        description: Host for server
      port:
        default: '8080'
        description: Port for server
      version:
        enum:
          - 'v1'
        default: 'v1'
```

File: openapi/api.yaml [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+paths:
+  /{version}/clusters:
```
Comment (rJankowski93) [inline openapi/api.yaml:9]:
```suggestion
  /clusters:
```

File: openapi/api.yaml [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    put:
+      description: update exisiting cluster
+      parameters:
+        - name: version
+          required: true
+          in: path
+          schema:
+            type: string
```
Comment (rJankowski93) [inline openapi/api.yaml:17]:
```suggestion

```

and the same for every version in path

Comment (dbadura) [reply openapi/api.yaml]: fixed

File: openapi/api.yaml (hunk 2 above)
Comment (dbadura) [reply openapi/api.yaml]: fixed

File: openapi/api.yaml (hunk 1 above)
Comment (dbadura) [reply openapi/api.yaml]: fixed

File: Makefile [hunk 4]
```diff
@@ -71,8 +71,28 @@ test-all: test
 clean:
 	rm -rf bin
 
+export OAPI_VALIDATOR=spectral lint
```
Comment (dbadura) [inline Makefile:74]: `spectral` is enought, move lint somewhere else

Comment (dbadura) [inline Makefile:74]: `spectral` is enough, move lint somewhere else

File: cmd/mothership/start/http.go [hunk 5]
```diff
@@ -60,6 +59,7 @@ func startWebserver(ctx context.Context, o *Options) error {
 		callHandler(o, statusChanges)).
 		Methods("GET")
 
+	//TODO: should we move it to other package as separate API?
```
Comment (dbadura) [inline cmd/mothership/start/http.go:62]: Remove it

File: configs/component-reconcilers.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-    "base": {
-        "url": "http://localhost:8081/v1/run"
+    "busola-migrator": {
```
Comment (dbadura) [inline configs/component-reconcilers.json:2]: revert it

File: pkg/reconciler/heartbeat/heartbeat.go [hunk 7]
```diff
@@ -87,9 +87,14 @@ func (su *Sender) sendUpdate(status reconciler.Status, reason error, onlyOnce bo
 	su.stopJob() //ensure previous interval-loop is stopped before starting a new loop
 
 	task := func(status reconciler.Status, rootCause error) error {
+		var errMsg *string
```
Comment (dbadura) [inline pkg/reconciler/heartbeat/heartbeat.go:90]: use `GetErrMessage` fn

File: pkg/reconciler/instances/busolamigrator/busola.go [hunk 8]
```diff
@@ -32,7 +32,7 @@ func init() {
 	//configure reconciler
 	reconciler.
 		//list dependencies (these components have to be available before this component reconciler is able to run)
-		WithDependencies("istio").
+		//WithDependencies("istio").
```
Comment (dbadura) [inline pkg/reconciler/instances/busolamigrator/busola.go:35]: revert it

File: scripts/git-check.sh [hunk 9]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (dbadura) [inline scripts/git-check.sh:1]: rename it to something better and add message about success

**Review by rJankowski93 — DISMISSED** [review]

File: openapi/README.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+## Swagger Editor
```
Comment (klaudiagrz) [inline openapi/README.md:1]:
```suggestion
## Show API specs in Swagger Editor
```

File: openapi/README.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+To be able to show Open API specs from several files without error in Swagger Editor, you have to:
```
Comment (klaudiagrz) [inline openapi/README.md:3]:
```suggestion
To successfully show Open API specs from several files in Swagger Editor, you have to: 

```

File: openapi/README.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
+## Swagger Editor
+
+To be able to show Open API specs from several files without error in Swagger Editor, you have to: 
+1. Run swagger using `docker-compose.yaml`.
```
Comment (klaudiagrz) [inline openapi/README.md:4]:
```suggestion
1. Run Swagger using `docker-compose.yaml`:
```

File: openapi/README.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+   ```bash
+   docker-compose up
+   ```
+
+2. Go to Swagger edit URL:`http://localhost:8081/?url=/docs/internal_api.yaml`
```
Comment (klaudiagrz) [inline openapi/README.md:9]:
```suggestion
2. Go to Swagger edit URL:
`http://localhost:8081/?url=/docs/internal_api.yaml`
```

File: openapi/README.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+
+Here is the [issue](https://github.com/swagger-api/swagger-editor/issues/1409) about error.
```
Comment (klaudiagrz) [inline openapi/README.md:12]:
```suggestion
For reference, see the [issue](https://github.com/swagger-api/swagger-editor/issues/1409) related to Swagger not being able to show specs from several files.
```

**Review by rJankowski93 — APPROVED** [review]

**Pull request merged**